### PR TITLE
feat(design): update quantity field focus functionality

### PR DESCRIPTION
**Pull request by xelaint**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes:

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Review by damienwebdev — APPROVED** [review]
lgtm

File: libs/design/src/atoms/form/quantity-field/quantity-field.component.ts [hunk 1]
```diff
@@ -55,7 +55,15 @@ export class DaffQuantityFieldComponent implements ControlValueAccessor, DaffFor
...
   @Input() selectMax = 10;
 
-  focused = false;
+  get focused() {
+    if (this.input || this.select) {
+      return this.input?.focused || this.select?.focused;
+    }
+
+    // If we don't have a control, quantity field should not be focused.
+    return false;
```
Comment (griest024) [inline libs/design/src/atoms/form/quantity-field/quantity-field.component.ts:64]:
```ts
return this.input?.focused || this.select?.focused
```
should be an equivalent implementation. Additionally there should be tests for this behavior.

File: libs/design/src/atoms/form/quantity-field/quantity-field.component.ts [hunk 2]
```diff
@@ -55,7 +55,15 @@ export class DaffQuantityFieldComponent implements ControlValueAccessor, DaffFor
    */
   @Input() selectMax = 10;
 
-  focused = false;
+  get focused() {
```
Comment (griest024) [inline libs/design/src/atoms/form/quantity-field/quantity-field.component.ts:58]: Type the getter function return: `get focused(): boolean {`

File: libs/design/src/atoms/form/quantity-field/quantity-input/quantity-input.component.ts [hunk 3]
```diff
@@ -44,8 +44,11 @@ export class DaffQuantityInputComponent implements OnInit, OnDestroy {
    * Instead, we listen for the change event and manually patch form control values.
    */
   _inputControl = new FormControl();
-  // TODO: should this not use this.input as the source of truth for focused?
-  focused = false;
+
+  get focused() {
```
Comment (griest024) [inline libs/design/src/atoms/form/quantity-field/quantity-input/quantity-input.component.ts:48]: Type the getter function return: `get focused(): boolean {`

**Review by griest024 — CHANGES_REQUESTED** [review]
lmk if I should make these changes

File: libs/design/src/atoms/form/quantity-field/quantity-field.component.ts (hunk 2 above)
Comment (xelaint) [reply libs/design/src/atoms/form/quantity-field/quantity-field.component.ts:58]: updated

File: libs/design/src/atoms/form/quantity-field/quantity-input/quantity-input.component.ts (hunk 3 above)
Comment (xelaint) [reply libs/design/src/atoms/form/quantity-field/quantity-input/quantity-input.component.ts:48]: updated

**Review by griest024 — APPROVED** [review]

**Review by griest024 — DISMISSED** [review]

**Review by griest024 — APPROVED** [review]

**Pull request merged**